CONTRIBUTING.md

```diff
@@ -0,0 +1,338 @@
+- `🐛 FIX:` (`:bug:`) — to fix a code bug.
+- `📖  DOCS:` (`:book:`) — to add new documentation.
+- `✏️ REWRITE:` (`:pencil2:`) — to apply fixes (like correcting typos) to existing documentation.
+- `🧪  TEST:` (`:testube:`) — to add additional testing only.
+- `🚀 RELEASE:` (`:rocket:`) — to bump the package version for release.
+- `⬆️ UPGRADE:` (`:arrow_up:`) — for upgrading a dependency pinning.
+- `♻️ REFACTOR:` (`:recycle:`) — for refactoring existing code (with no specific improvements).
+- `🗑️ DEPRECATE:` (`:wastebasket:`) — mark some code as deprecated (for removal in a later release). The future version when it will be removed should also be specified, and (if applicable) what will replace it.
+- `🔀 MERGE:` (`:twisted_rightwards_arrows:`) — for a merge commit (then all commits within the merge should be categorised)
+- `❓ OTHER:` (`:question:`) — anything not covered above (use as a last resort!).
+
+This list is loosely in order of priority, e.g. a commit that is both a bug fix and back-incompatible should be categorised as `BREAKING` not `FIX`.
+
+(dev/releases)=
+
+## Releases and Change-logs
+
+Releases should be made *via* [GitHub Releases](https://docs.github.com/en/github/administering-a-repository/managing-releases-in-a-repository), from the `master` branch and using [semantic versioning](https://semver.org/) for tags, e.g. `v1.2.1`, **for versions above 1.0.0**.
+For versions below 1.0.0, it is understood that breaking changes are more frequent (i.e. the repo is in beta), and so semantic versioning is relaxed such that MINOR version changes also signify backward incompatible releases.
+
+The change-log should be easy for users and developers to understand the key changes (as [discussed here](https://keepachangelog.com/)), and should mirror the commits categories described above, with the following format:
+
+```md
+## 1.1.0 - 2020-06-25
+
+### Added
+- List of `NEW` commits
+
+### Improved
+- List of `IMPROVE` commits
+
+### Fixed
+- List of `FIX` commits
+
+### Breaking
+- List of `BREAKING` and `UPGRADE` commits
+
+### Deprecated
+- List of `DEPRECATE` commits
+
+### Documented
+- List of `DOCS` commits
+
+```
+
+Sub-headings with no content can be skipped and commits by contributors should be given attrition (e.g. ", thanks to @chrisjsewell").
+
+Package releases (e.g. to PyPi or npm) should be automated *via* GitHub Action workflows, triggered on tag creation.
+
+The full workflow would then be:
+
+- Make a release commit `🚀 RELEASE: ...` on `master` (*via* PR) which bumps the version to `M.m.p` (e.g. changing `__version__` for python packages) and adds a section to `CHANGELOG.md` in the format above.
+- Create a GitHub release for that commit, with tag `vM.m.p`, heading `Version M.m.p` (optionally including the changelog section in the body).
+- Check that automated release workflows complete successfully.
+
+(dev/deprecations)=
+
+## Deprecations
```

Comment:
Should we note that we also follow the `semver` approach of "anything is fair game before version `1.0`?

Reply:
Added in 9487cf83e0287f72125bf56733759f86b0bf499e